lib/resolve-package-path.js

```diff
@@ -194,11 +194,14 @@ module.exports = function resolvePackagePath(caches, name, dir) {
   // Ensure that basedir is an absolute path at this point.
   var absoluteStart = path.resolve(basedir);
 
-  // convert absoluteStart to a real path (follow links, etc).
-  absoluteStart = _getRealDirectoryPath(caches.REAL_DIRECTORY_PATH, absoluteStart);
+  while (_getRealDirectoryPath(caches.REAL_DIRECTORY_PATH, absoluteStart) === null) {
+    absoluteStart = path.dirname(absoluteStart);
+  }
 
   if (!absoluteStart) {
-    throw new TypeError('resolvePackagePath: \'dir\' must refer to a valid directory name.');
+    var error = new TypeError('resolvePackagePath: \'dir\' must refer to a valid directory name.');
```

Comment:
this error message will need to be adjusted, and additional tests will need to be written.